pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -166,61 +195,66 @@ func (r *ReconcileDependencyBuild) handleStateSubmitBuild(ctx context.Context, d
 	}
 	//now we submit the build
 	if err := r.client.Create(ctx, &tr); err != nil {
+		if errors.IsAlreadyExists(err) {
+			log.V(4).Info("handleStateBuilding: taskrun %s:%s already exists, not retrying", tr.Namespace, tr.Name)
+			return reconcile.Result{}, nil
+		}
 		r.eventRecorder.Eventf(db, v1.EventTypeWarning, "TaskRunCreationFailed", "The DependencyBuild %s/%s failed to create its build pipeline run", db.Namespace, db.Name)
 		return reconcile.Result{}, err
 	}
+
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileDependencyBuild) handleStateBuilding(ctx context.Context, depId string, db *v1alpha1.DependencyBuild) (reconcile.Result, error) {
-	//make sure we still have a linked pr
-	list := &pipelinev1beta1.TaskRunList{}
-	lbls := map[string]string{
-		artifactbuild.DependencyBuildIdLabel: depId,
-	}
-	listOpts := &client.ListOptions{
-		Namespace:     db.Namespace,
-		LabelSelector: labels.SelectorFromSet(lbls),
-	}
-	if err := r.nonCachingClient.List(ctx, list, listOpts); err != nil {
-		return reconcile.Result{}, err
-	}
-	if len(list.Items) == 0 {
-		//no linked pr, this seems to happen randomly where the PR does not show up
-		//just return and next reconcile it is there
-		//TODO: Why is this happening? caching?
-		r.eventRecorder.Eventf(db, v1.EventTypeWarning, "NoTaskRun", "The DependencyBuild %s/%s did not have any TaskRuns", db.Namespace, db.Name)
-		return reconcile.Result{}, nil
-	}
-	var pr *pipelinev1beta1.TaskRun
-	//look for the most recent one, there could be multiple builds if earlier recipes failed
-	for _, current := range list.Items {
-		if pr == nil || pr.CreationTimestamp.Before(&current.CreationTimestamp) {
-			pr = &current
+func (r *ReconcileDependencyBuild) handleTaskRunReceived(ctx context.Context, tr *pipelinev1beta1.TaskRun) (reconcile.Result, error) {
+	if tr.Status.CompletionTime != nil {
+		// get db
+		ownerRefs := tr.GetOwnerReferences()
+		if ownerRefs == nil || len(ownerRefs) == 0 {
+			msg := "taskrun missing onwerrefs %s:%s"
+			r.eventRecorder.Eventf(tr, v1.EventTypeWarning, msg, tr.Namespace, tr.Name)
+			log.Info(msg, tr.Namespace, tr.Name)
+			return reconcile.Result{}, nil
 		}
-	}
-	if pr.Status.CompletionTime != nil {
-		if pr.Name == db.Status.LastCompletedBuildTaskRun {
+		if len(ownerRefs) > 1 {
```

Comment:
looking at the artifactbuild flow now, I think I now see that a taskrun could have both a db and abr owner ref

assuming that can occur at the same time for a given taskrun, I need to distinguish based on ownerref kind above here @stuartwdouglas , right?

that said, as I said before, whatever negative effect this might have on the test run, I was unable to discern it (though that most likely is my not having sufficient understanding of how to analyze the results